examples/ticks_and_spines/tick_transform_formatter.py

```diff
@@ -0,0 +1,128 @@
+     ensure that the inputs are indeed integers.
+
+The same histogram is plotted in two sub-plots with a shared x-axis.
+Each axis shows a different temperature scale: one in degrees Celsius,
+one in degrees Rankine (the Fahrenheit analogue of Kelvins). This is one
+of the few examples of recognized scientific units that have both a
+scale and an offset relative to each other.
+"""
+
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib.axis import Ticker
+from matplotlib.ticker import (
+    TransformFormatter, StrMethodFormatter, MaxNLocator
+)
+
+
+class LinearTransform:
+    """
+    A callable class that transforms input values to output according to
+    a linear transformation.
+    """
+
+    def __init__(self, in_start=None, in_end=None, out_start=None, out_end=None):
+        """
+        Sets up the transformation such that `in_start` gets mapped to
+        `out_start` and `in_end` gets mapped to `out_end`. The following
+        shortcuts apply when only some of the inputs are specified:
+
+          - none: no-op
+          - in_start: translation to zero
+          - out_start: translation from zero
+          - in_end: scaling to one (divide input by in_end)
+          - out_end: scaling from one (multiply input by in_end)
+          - in_start, out_start: translation
+          - in_end, out_end: scaling (in_start and out_start zero)
+          - in_start, out_end: in_end=out_end, out_start=0
+          - in_end, out_start: in_start=0, out_end=in_end
+
+        Based on the following rules:
+
+          - start missing: set start to zero
+          - both ends are missing: set ranges to 1.0
+          - one end is missing: set it to the other end
+        """
+        self._in_offset = 0.0 if in_start is None else in_start
+        self._out_offset = 0.0 if out_start is None else out_start
+
+        if in_end is None:
+            if out_end is None:
+                self._in_scale = 1.0
+            else:
+                self._in_scale = out_end - self._in_offset
+        else:
+            self._in_scale = in_end - self._in_offset
+
+        if out_end is None:
+            if in_end is None:
+                self._out_scale = 1.0
+            else:
+                self._out_scale = in_end - self._out_offset
+        else:
+            self._out_scale = out_end - self._out_offset
+
+    def __call__(self, x):
+        """
+        Transforms the input value `x` according to the rule set up in
+        `__init__`.
+        """
+        return ((x - self._in_offset) * self._out_scale / self._in_scale +
+                     self._out_offset)
+
+# X-data
+temp_C = np.arange(-5.0, 5.1, 0.25)
+# Y-data
+counts = 15.0 * np.exp(-temp_C**2 / 25)
+# Add some noise
+counts += np.random.normal(scale=4.0, size=counts.shape)
+if counts.min() < 0:
+    counts += counts.min()
+
+fig = plt.figure()
+ax1 = fig.add_subplot(111)
+ax2 = fig.add_subplot(111, sharex=ax1, sharey=ax1, frameon=False)
```

Comment:
Not strictly necessary, but the conflation was deliberate. I wanted to be able to display both the original and the transformed values as well as to use this as a demo for #7528. There will be some modifications once that PR gets accepted (which I think is very likely once I find the time to write some additional tests for it).